cluster-autoscaler/core/scaledown/actuation/drain.go

```diff
@@ -43,106 +42,90 @@ const (
+	// is killed with SIGKILL after MaxGracefulTerminationTime/ShutdownGracePeriodSeconds
 	DefaultPodEvictionHeadroom = 30 * time.Second
-	// DefaultDsEvictionEmptyNodeTimeout is the time to evict all DaemonSet pods on empty node
-	DefaultDsEvictionEmptyNodeTimeout = 10 * time.Second
-	// DefaultDsEvictionRetryTime is a time between retries to create eviction that uses for DaemonSet eviction for empty nodes
-	DefaultDsEvictionRetryTime = 3 * time.Second
 )
 
 type evictionRegister interface {
 	RegisterEviction(*apiv1.Pod)
 }
 
-// Evictor can be used to evict pods from nodes.
 type Evictor struct {
-	EvictionRetryTime          time.Duration
-	DsEvictionRetryTime        time.Duration
-	DsEvictionEmptyNodeTimeout time.Duration
-	PodEvictionHeadroom        time.Duration
-	evictionRegister           evictionRegister
-	deleteOptions              simulator.NodeDeleteOptions
+	EvictionRetryTime                time.Duration
+	PodEvictionHeadroom              time.Duration
+	evictionRegister                 evictionRegister
+	shutdownGracePeriodByPodPriority []kubelet_config.ShutdownGracePeriodByPodPriority
+	priorityEvictorEnabled           bool
 }
 
-// NewDefaultEvictor returns an instance of Evictor using the default parameters.
-func NewDefaultEvictor(deleteOptions simulator.NodeDeleteOptions, evictionRegister evictionRegister) Evictor {
+// NewEvictor returns an instance of Evictor.
+func NewEvictor(evictionRegister evictionRegister, shutdownGracePeriodByPodPriority []kubelet_config.ShutdownGracePeriodByPodPriority, priorityEvictorEnabled bool) Evictor {
+	sort.Slice(shutdownGracePeriodByPodPriority, func(i, j int) bool {
+		return shutdownGracePeriodByPodPriority[i].Priority < shutdownGracePeriodByPodPriority[j].Priority
+	})
+
 	return Evictor{
-		EvictionRetryTime:          DefaultEvictionRetryTime,
-		DsEvictionRetryTime:        DefaultDsEvictionRetryTime,
-		DsEvictionEmptyNodeTimeout: DefaultDsEvictionEmptyNodeTimeout,
-		PodEvictionHeadroom:        DefaultPodEvictionHeadroom,
-		evictionRegister:           evictionRegister,
-		deleteOptions:              deleteOptions,
+		EvictionRetryTime:                DefaultEvictionRetryTime,
+		PodEvictionHeadroom:              DefaultPodEvictionHeadroom,
+		evictionRegister:                 evictionRegister,
+		shutdownGracePeriodByPodPriority: shutdownGracePeriodByPodPriority,
+		priorityEvictorEnabled:           priorityEvictorEnabled,
 	}
 }
 
-// DrainNode works like DrainNodeWithPods, but lists of pods to evict don't have to be provided. All non-mirror, non-DS pods on the
-// node are evicted. Mirror pods are not evicted. DaemonSet pods are evicted if DaemonSetEvictionForOccupiedNodes is enabled, or
-// if they have the EnableDsEvictionKey annotation.
-func (e Evictor) DrainNode(ctx *acontext.AutoscalingContext, nodeInfo *framework.NodeInfo) (map[string]status.PodEvictionResult, error) {
-	dsPodsToEvict, nonDsPodsToEvict := podsToEvict(ctx, nodeInfo)
```

Comment:
Since ```DrainNode``` and ```EvictDaemonSetPods``` differ by ```DaemonSetEvictionForOccupiedNodes``` and ```DaemonSetEvictionForEmptyNodes``` flags, added back the ```EvictDaemonSetPods``` and changed the function arguments to accept ```nodeInfo```